tests/unit/design_document_tests.py

```diff
@@ -841,6 +841,43 @@ def test_get_search_info(self):
         self.assertTrue(search_index_metadata['pending_seq'] <= 101, 'The pending_seq should be 101 or fewer.')
         self.assertTrue(search_index_metadata['disk_size'] >0, 'The disk_size should be greater than 0.')
 
+    @unittest.skipUnless(
+        os.environ.get('RUN_CLOUDANT_TESTS') is not None,
+        'Skipping Cloudant _search_info endpoint test'
+    )
+    def test_get_search_disk_size(self):
+        """
+        Test retrieval of search_disk_size endpoint from the DesignDocument.
+        """
+        self.populate_db_with_documents(100)
+        ddoc = DesignDocument(self.db, '_design/ddoc001')
+        ddoc.add_search_index(
+            'search001',
+            'function (doc) {\n  index("default", doc._id); '
+            'if (doc._id) {index("name", doc.name, {"store": true}); }\n}'
+        )
+        ddoc.save()
+
+        ddoc_remote = DesignDocument(self.db, '_design/ddoc001')
+        ddoc_remote.fetch()
+
+        ddoc_remote.search_info('search001')  # trigger index build
+
+        search_disk_size = ddoc_remote.search_disk_size('search001')
+
+        self.assertEquals(
+            search_disk_size.keys(), ['name', 'search_index'],
+            'The search disk size should contain only keys "name" and "search_index"')
```

Comment:
Easier to understand test failure if you dump out the actual keys. `assertItemsEqual` might actually do this for you too.